core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -163,6 +163,30 @@ abstract class AppBase(
     }
   }
 
+  /**
+   * Calculates total core ms which is the sum over executor core ms. Core ms per executor
+   * is computed as executor duration (ms) multiplied by num of cores in the executor.
+   */
+  def calculateTotalCoreMs(): Long = {
+    var totalCoreMs: Long = 0L
+    executorIdToInfo.foreach { case(_, eInfo) =>
+      val eStartTime = eInfo.addTime.getTime
+      var eEndTime = eInfo.removeTime
```

Comment:
@parthosa and @cindyyuanjiang Out of curiosity, why do we have addTime defined as date while endTime is defined as long?
Is there a good technical reason to save addTime as date? Convert from tolong to Date, then once more to Long could raise some problems with date format/calendar.
If no need to keep the addTime as date, then it should be also a Long

Reply:
I think this has been present historically. Based on the current usages, I dont see any reason to keep them of different types.

https://github.com/NVIDIA/spark-rapids-tools/blob/1564e71d202470950a507229212e653e4093338b/core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ClassWarehouse.scala#L45-L46

Reply:
+1, I can file a followup issue, or we can change in this PR, WDYT @amahussein?

Reply:
Correct me if I am wrong, I don't see changing the type of `addTime` will have an impact on output files.
If that's case let's fix the type in this PR to save the overhead of a new followup just to change 2 lines.

Reply:
@amahussein updated, thanks!